lib/tai/advisors/factories/shared_venues_and_shared_products.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Tai.Advisors.Factories.SharedVenuesAndSharedProducts do
+  @moduledoc """
+  Advisory factory for sharing all subscribed products. Advisor name falls back to `group.id`.
+
+  Use this to receive order book updates from all subscribed products in that group.
+  """
+  @behaviour Tai.Advisors.Factory
+
+  @type group :: Tai.AdvisorGroup.t()
+  @type spec :: Tai.Advisor.spec()
+
+  @spec advisor_specs(group) :: [spec]
+  def advisor_specs(group) do
+    opts = [
+      group_id: group.id,
+      advisor_id: group.id,
+      products: group.products,
+      config: group.config
+    ]
+
+    [{group.advisor, opts}]
```

Comment:
Can you add a test for this factory. e.g. https://github.com/fremantle-capital/tai/blob/master/test/tai/advisors/factories/one_per_venue_and_product_test.exs